tools/rostopic/src/rostopic/__init__.py

```diff
@@ -177,39 +180,93 @@ def print_hz(self):
-    if filter_expr is not None:
-        # have to subscribe with topic_type
-        sub = rospy.Subscriber(real_topic, msg_class, rt.callback_hz)
-    else:
-        sub = rospy.Subscriber(real_topic, rospy.AnyMsg, rt.callback_hz)        
-    print("subscribed to [%s]"%real_topic)
+    hz_checkers = []
+    for topic in topics:
+        msg_class, real_topic, _ = get_topic_class(topic, blocking=True) # pause hz until topic is published
+        rt = ROSTopicHz(window_size, filter_expr=filter_expr, use_wtime=use_wtime)
+        # we use a large buffer size as we don't know what sort of messages we're dealing with.
+        # may parameterize this in the future
+        if filter_expr is not None:
+            # have to subscribe with topic_type
+            rospy.Subscriber(real_topic, msg_class, rt.callback_hz)
+        else:
+            rospy.Subscriber(real_topic, rospy.AnyMsg, rt.callback_hz)
+        hz_checkers.append(rt)
+        print("subscribed to [%s]" % real_topic)
 
     if rospy.get_param('use_sim_time', False):
         print("WARNING: may be using simulated time",file=sys.stderr)
 
     while not rospy.is_shutdown():
         _sleep(1.0)
-        rt.print_hz()
+        if len(topics) == 1:
+            hz_checkers[0].print_hz()
+            continue
+        # monitoring multiple topics' hz
+        header = ['topic', 'rate', 'min_delta', 'max_delta', 'std_dev', 'window']
+        stats = {h: [] for h in header}
+        if not any(rt.times for rt in hz_checkers):
+            continue  # wait for initial message
+        for topic, rt in zip(topics, hz_checkers):
+            hz_stat = rt.get_hz()
+            if hz_stat is None:
+                continue
+            rate, min_delta, max_delta, std_dev, window = hz_stat
+            stats['topic'].append(topic)
+            stats['rate'].append('{:.4}'.format(rate))
+            stats['min_delta'].append('{:.4}'.format(min_delta))
+            stats['max_delta'].append('{:.4}'.format(max_delta))
+            stats['std_dev'].append('{:.4}'.format(std_dev))
+            stats['window'].append(str(window))
+        if len(stats['topic']) == 1:
+            print('no new messages')
+            continue
+        # compose table with left alignment
+        header_aligned = []
+        col_widths = []
+        for h in header:
+            col_width = max(len(h), max(len(el) for el in stats[h]))
+            col_widths.append(col_width)
+            header_aligned.append(h.center(col_width))
+            for i, el in enumerate(stats[h]):
+                stats[h][i] = stats[h][i].ljust(col_width)
+        # sum of col and each 3 spaces width
+        table_width = sum(col_widths) + 3 * (len(header) - 1)
+        body = '\n'.join('   '.join(stats[h][i] for h in header) for i in xrange(len(stats['topic'])))
+        table = '{header}\n{hline}\n{body}\n'.format(
+            header='   '.join(header_aligned), hline='=' * table_width, body=body)
+        print(table)
```

Comment:
Should this be done by `print_hz` too? the function could have a verbose mode (used for printing a single topic) and a "single line" mode for this case.